services/kubernetesbuilddeploymonitor/src/index.ts

```diff
@@ -181,6 +181,13 @@ ${podLog}`;
       }
     })(jobInfo.status.active ? 'active' : jobInfo.status.conditions[0].type.toLowerCase());
 
+    if (status == 'failed') {
+      // failed jobs in kubernetes don't have a completionTime in them
+      // the status conditions will have a lastTransitionTime in them that will contain the state the job went into
+      // since jobs only have 1 attempt, there shouldn't be any other conditions
+      jobInfo.status.completionTime = jobInfo.status.conditions[0].lastTransitionTime
```

Comment:
This mutates the `jobInfo` which is considered bad practice in JS these days. It might cause some poor, unsuspecting, future developer (karma says it's you 😉) to check this field at some point later on in this file and assume something that isn't true.

It'd be better to create a new variable that's set with the `jobInfo.status.completionTime` or `jobInfo.status.conditions[0].lastTransitionTime` to use in the `updateDeployment`. Similar to `status` right above (but doesn't need a fancy anonymous function).

Reply:
Sure, easy fix